k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/selectedstate/command/ImapCommandSplitter.java

```diff
@@ -0,0 +1,106 @@
+package com.fsck.k9.mail.store.imap.selectedstate.command;
+
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+
+import com.fsck.k9.mail.store.imap.selectedstate.command.FolderSelectedStateCommand.Builder;
+import com.fsck.k9.mail.store.imap.selectedstate.command.FolderSelectedStateCommand.ContiguousIdGroup;
+
+
+class ImapCommandSplitter {
+
+    static List<FolderSelectedStateCommand> splitCommand(FolderSelectedStateCommand command, int lengthLimit) {
+        List<FolderSelectedStateCommand> commands = new ArrayList<>();
+
+        if (command.getIdSet() != null || command.getIdGroups() != null) {
```

Comment:
if you invert this `if` and throw, you can save one indentation level for the rest of the method